Add item type labels to config documentation

alistair3149 opened the pull request:
Closes: #198

Key changes:
- Add item type label section
- Refactor config documentation into Mustache
- Fix minor typo

![image](https://github.com/user-attachments/assets/02667f7d-177a-4fe4-a4da-e9d93e27a5cb)

JeroenDeDauw commented:
Product thoughts on `item type labels` section:

![image](https://github.com/user-attachments/assets/f677784c-e97b-49c4-97e9-3c9d604ef05f)

* Change column header from `ID` to `Item`
* Show the Item's Wikibase Label in brackets next to the ID
* Link the entire text to the page (wiki/Item:Q123)
* The column header `Label` is confusing, since it is not the Wikibase Label. Same for the section header and the description: we should not call it "label". Something like "Tab header text" or "Tab name" solves the confusion issue by being distinct, and also makes it obvious what the thing is for.

alistair3149 commented:
> Having the table with edit links is great. Putting it in the documentation might not be optimal. It's somewhat surprising to have to scroll through docs to find a UI for editing some specific config. Could move this UI up, though unsure if that works out better.
> 
> Things would be less weird if we had a reactive UI for the entire config (rather than editing the JSON directly), but that is a bunch of work to create.

As a stopgap solution, we can add an anchor link that links to the table to the top of the configuration page. We can revisit the reactive UI solution in the future

malberts commented on src/Presentation/ConfigEditPageTextBuilder.php:
```diff
@@ -124,4 +49,31 @@ private function getExampleContents(): string {
 	}
 
+	private function getItemTypesData(): array {
+		$itemTypes = $this->config->getItemTypes();
+		if ( $itemTypes === [] ) {
+			return [];
+		}
```
I think in cases like this it is not necessary to have an early exit. If the logic is based on an array loop, then the loop will do nothing and return an empty array anyway. The early exit suggests the empty array needs to be handled explicitly.

This can also be seen if the logic here was replaced by something like:
```php
return array_map(
    fn( $itemType ) => $this->getItemTypeData( $itemType ),
    $itemTypes
);
```

malberts commented on src/Presentation/ConfigEditPageTextBuilder.php:
```diff
@@ -124,4 +49,31 @@ private function getExampleContents(): string {
+		$data = [];
+		foreach ( $itemTypes as $itemType ) {
+			$id = $itemType->getSerialization();
+			if ( !$id ) {
```
This check should be explicit. However, why do we need to check for this? The `ItemId` constructor already checks that the format is valid. So the `ItemId`s you get from the config are already valid.

malberts commented:
Regarding the table position: I also think we should move it directly under the form, above the documentation. The table is part of the configuration, not documentation, so should be grouped with the rest of the config.

alistair3149 commented:
I have addressed all the issues in code review and also:
- Add i18n for all parts of the table
- Use `LinkRenderer` for proper handling for URLs

I need some refactor suggestions on breaking down the class further since there might be too much dependencies in the constructor.

JeroenDeDauw commented:
> I need some refactor suggestions on breaking down the class further since there might be too much dependencies in the constructor.

You could extract the code that builds the table. That has its own set of dependencies and does not need the template renderer. This also naturally exposes the view model it creates via a public method, making testing easier.
